app/RouteHandlers/SofortNotificationHandler.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\App\RouteHandlers;
+
+use DateTime;
+use Psr\Log\LogLevel;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use WMDE\Fundraising\Frontend\Factories\FunFunFactory;
+use WMDE\Fundraising\Frontend\PaymentContext\RequestModel\SofortNotificationRequest;
+use WMDE\Fundraising\Frontend\PaymentContext\ResponseModel\SofortNotificationResponse;
+
+class SofortNotificationHandler {
+
+	private $ffFactory;
+
+	public function __construct( FunFunFactory $ffFactory ) {
+		$this->ffFactory = $ffFactory;
+	}
+
+	public function handle( Request $request ): Response {
+		$useCase = $this->ffFactory->newHandleSofortPaymentNotificationUseCase( $request->query->get( 'updateToken' ) );
+
+		$response = $useCase->handleNotification( $this->newUseCaseRequestFromRequest( $request ) );
+
+		$this->logResponseIfNeeded( $response, $request );
+
+		if ( !$response->notificationWasHandled() ) {
+			return new Response( 'Bad request', Response::HTTP_BAD_REQUEST );
+		} elseif ( $response->hasErrors() ) {
+			return new Response( 'Error', Response::HTTP_INTERNAL_SERVER_ERROR );
+		}
+
+		return new Response( 'Ok', Response::HTTP_OK );
+	}
+
+	private function newUseCaseRequestFromRequest( Request $request ): SofortNotificationRequest {
+		$useCaseRequest = new SofortNotificationRequest();
+		$useCaseRequest->setDonationId( $request->query->getInt( 'id' ) );
+		$useCaseRequest->setTransactionId( $request->request->get( 'transaction', '' ) );
+
+		$time = DateTime::createFromFormat( DateTime::ATOM, $request->request->get( 'time', '' ) );
+
+		if ( $time instanceof DateTime ) {
+			$useCaseRequest->setTime( $time );
+		}
```

Comment:
@gbirke told me about this and that the error should probably be handled here and not in the UC. I fully agree with this, as what we are catching here is a parsing error, which is presentation specific. However this PR is already quite big, has been sitting here for multiple days and it is quite annoying to have to deal with the whole thing if only small stuff gets changed. So I suggest we merge as is and do a follow up. I'm happy to make that follow up and will do so if no one beats me to it. This will include having a good look at the logging as well, which I think does not belong in this Handler, and could perhaps be done in a way that makes it easier to test (no e2e needed).